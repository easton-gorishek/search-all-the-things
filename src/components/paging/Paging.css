:local(.paging) {
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
  font-size: 14px;

  display: flex;
  justify-content: center;
  align-items: center;

  button {
    font-size: 15px;
    border-radius: 5px;
    margin: 5px;
  }
}