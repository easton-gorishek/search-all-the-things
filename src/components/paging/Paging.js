import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Paging.css';

export default class Paging extends Component {

  static propTypes = {
    page: PropTypes.number,
    results: PropTypes.arrayOf(Array),
    onPage: PropTypes.func.isRequired
  };

  handlePage(increment) {
    const { page, onPage, results } = this.props;
    if(page >= 1 || page <= results.length) {
      onPage(page + increment);
    }
  }

  render() {
    const { page, results } = this.props;
    if(!results) return null;
    
    return (
      <div className={styles.paging}>
        <button disabled={page === 1} onClick={() => this.handlePage(-1)}>Previous</button>
        <p>Page  {page} of {results.length}</p>
        <button disabled={page === results.length} onClick={() => this.handlePage(+1)}>Next</button>
      </div>
    );
  }
}
