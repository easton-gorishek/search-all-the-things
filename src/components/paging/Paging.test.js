import React from 'react';
import { shallow } from 'enzyme';
import Paging from './Paging';
import toJSON from 'enzyme-to-json';

describe('Paging', () => {

  it('renders as designed', () => {
    // shallow render is "visualization of what got render"
    const wrapper = shallow(<Paging onSearch={() => {}}/>);
    // make sure it is still doing the same thing
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});