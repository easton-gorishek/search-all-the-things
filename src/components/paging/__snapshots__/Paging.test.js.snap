// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging renders as designed 1`] = `""`;
